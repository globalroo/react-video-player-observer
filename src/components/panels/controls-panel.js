import React from "react";
import PropTypes from "prop-types";

export const createClickHandler = ({ loadAsset, asset }) => () => loadAsset(asset);

export const MovieButton = ({ asset, loadAsset }) => {
	const clickHandler = createClickHandler({ loadAsset, asset });
	return (
		<div className="btn-group" role="group">
			<button type="button" onClick={clickHandler} className="btn btn-info">{asset.name}</button>
		</div>
	);
};

MovieButton.propTypes = {
	loadAsset: PropTypes.func,
	asset: PropTypes.object
};

export const ControlsPanel = ({ availableMedia, loadAsset, asset }) => (
	<div className="panel panel-default">
		<div className="panel-heading">
			<h4>Controls</h4>
		</div>
		<div className="panel-body">
			<div className="btn-group btn-group-vertical btn-group-max-width" role="group" aria-label="Switch movies">
				{
					availableMedia.map((asset, ix) =>
						<MovieButton key={`asset_${ix}`} asset={asset} loadAsset={loadAsset} />)
				}
			</div>
		</div>
		<div className="panel-footer">
			{(asset && asset.name) && <small>Now playing: <span className="now-playing">{asset.name}</span></small>}
		</div>
	</div>
);

ControlsPanel.propTypes = {
	availableMedia: PropTypes.array,
	loadAsset: PropTypes.func,
	asset: PropTypes.object
};
