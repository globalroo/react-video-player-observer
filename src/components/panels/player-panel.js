import { Player } from "src/components/player/player";
import React from "react";
import PropTypes from "prop-types";

export const PlayerPanel = ({ onPlayerLoaded }) => (
	<div className="panel panel-default">
		<div className="panel-heading">
			<h4>Video Player</h4>
		</div>
		<div className="panel-body">
			<Player onPlayerLoaded={onPlayerLoaded} />
		</div>
	</div>
);

PlayerPanel.propTypes = {
	onPlayerLoaded: PropTypes.func,
};
