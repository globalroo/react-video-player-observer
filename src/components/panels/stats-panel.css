.latestEvent {
	font-size:130%;
}
.halfColumn {
	width: 50%;
}
