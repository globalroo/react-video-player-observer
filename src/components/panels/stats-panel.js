import "./stats-panel.css";
import React from "react";
import PropTypes from "prop-types";

export const StatsPanel = ({ playerState, eventHistory }) => {
	const currentTime = (playerState && playerState.currentTime) ? playerState.currentTime.toFixed(3) : 0;
	const remainingTime = (playerState && playerState.remainingTime) ? playerState.remainingTime.toFixed(3) : 0;
	const stateClass = ( playerState.error.code !== 0) ? "danger" : "info";
	return (
		<div className="panel panel-default">
			<div className="panel-heading">
				<h4>Observer</h4>
			</div>
			<div className="panel-body">
				<table className="table table-striped table-responsive">
					<tbody>
						<tr>
							<td className="halfColumn">currentTime</td>
							<td className="halfColumn">{currentTime}</td>
						</tr>
						<tr>
							<td>remainingTime</td>
							<td>{remainingTime}</td>
						</tr>
						<tr>
							<td>error?</td>
							<td className={stateClass}>{playerState.error.description} [{playerState.error.code}]</td>
						</tr>
						{
							eventHistory.map(({ event, timeStamp }, ix) => {
								const className = (ix === 0) ? "success latestEvent" : "";
								return (
									<tr className={className} key={ix} >
										<td>{event}</td>
										<td>{timeStamp}</td>
									</tr>
								);
							})
						}
					</tbody>
				</table>
			</div>
		</div>
	);
};

StatsPanel.propTypes = {
	playerState: PropTypes.object,
	eventHistory: PropTypes.array
};
