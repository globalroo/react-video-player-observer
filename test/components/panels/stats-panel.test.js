import React from "react";
import * as renderer from "react-test-renderer";
import { StatsPanel } from "src/components/panels/stats-panel";

describe("Stats Panel tests", () => {
	test("Controls panel renders as expected when initialised", () => {
		const playerState = {
			currentTime: 0,
			remainingTime: 100,
			error: {
				code: 0,
				description: ""
			}
		};
		const eventHistory = [{ event: "test", timeStamp: "now" }];
		const component = renderer.create(
			<StatsPanel playerState={playerState} eventHistory={eventHistory}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test("Controls panel renders as expected when error state and player at non zero position", () => {
		const playerState = {
			currentTime: 49.500,
			remainingTime: 49.500,
			error: {
				code: 3,
				description: "Decoding error"
			}
		};
		const eventHistory = [{ event: "error", timeStamp: "now" },{ event: "test", timeStamp: "now" }];
		const component = renderer.create(
			<StatsPanel playerState={playerState} eventHistory={eventHistory}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	test("Controls panel renders as expected when asset completed", () => {
		const playerState = {
			currentTime: 100,
			remainingTime: 0,
			error: {
				code: 0,
				description: "OK"
			}
		};
		const eventHistory = [{ event: "test", timeStamp: "now" }];
		const component = renderer.create(
			<StatsPanel playerState={playerState} eventHistory={eventHistory}/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

});
