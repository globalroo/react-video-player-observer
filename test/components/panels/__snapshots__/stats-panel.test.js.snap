// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Panel tests Controls panel renders as expected when asset completed 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <h4>
      Observer
    </h4>
  </div>
  <div
    className="panel-body"
  >
    <table
      className="table table-striped table-responsive"
    >
      <tbody>
        <tr>
          <td
            className="halfColumn"
          >
            currentTime
          </td>
          <td
            className="halfColumn"
          >
            100.000
          </td>
        </tr>
        <tr>
          <td>
            remainingTime
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            error?
          </td>
          <td
            className="info"
          >
            OK
             [
            0
            ]
          </td>
        </tr>
        <tr
          className="success latestEvent"
        >
          <td>
            test
          </td>
          <td>
            now
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Stats Panel tests Controls panel renders as expected when error state and player at non zero position 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <h4>
      Observer
    </h4>
  </div>
  <div
    className="panel-body"
  >
    <table
      className="table table-striped table-responsive"
    >
      <tbody>
        <tr>
          <td
            className="halfColumn"
          >
            currentTime
          </td>
          <td
            className="halfColumn"
          >
            49.500
          </td>
        </tr>
        <tr>
          <td>
            remainingTime
          </td>
          <td>
            49.500
          </td>
        </tr>
        <tr>
          <td>
            error?
          </td>
          <td
            className="danger"
          >
            Decoding error
             [
            3
            ]
          </td>
        </tr>
        <tr
          className="success latestEvent"
        >
          <td>
            error
          </td>
          <td>
            now
          </td>
        </tr>
        <tr
          className=""
        >
          <td>
            test
          </td>
          <td>
            now
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Stats Panel tests Controls panel renders as expected when initialised 1`] = `
<div
  className="panel panel-default"
>
  <div
    className="panel-heading"
  >
    <h4>
      Observer
    </h4>
  </div>
  <div
    className="panel-body"
  >
    <table
      className="table table-striped table-responsive"
    >
      <tbody>
        <tr>
          <td
            className="halfColumn"
          >
            currentTime
          </td>
          <td
            className="halfColumn"
          >
            0
          </td>
        </tr>
        <tr>
          <td>
            remainingTime
          </td>
          <td>
            100.000
          </td>
        </tr>
        <tr>
          <td>
            error?
          </td>
          <td
            className="info"
          >
            
             [
            0
            ]
          </td>
        </tr>
        <tr
          className="success latestEvent"
        >
          <td>
            test
          </td>
          <td>
            now
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
